File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/termsandconditions/TermsAndConditionsService.java
package io.zenwave360.example.clinicaltool.modules.termsandconditions;

import io.zenwave360.example.clinicaltool.modules.termsandconditions.domain.*;
import io.zenwave360.example.clinicaltool.modules.termsandconditions.dtos.*;
import java.math.*;
import java.time.*;
import java.util.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/**
 * Inbound Service Port for managing [AcceptedTermsAndConditions, TermsAndConditions].
 */
@org.springframework.modulith.NamedInterface("TermsAndConditionsService")
public interface TermsAndConditionsService {

     /**
      * 
      *
      */

    public List<TermsAndConditions> listTermsAndConditions()
;
     /**
      * 
      *
      */

    public Optional<TermsAndConditions> getTermsAndConditions(Long id)
;
     /**
      * 
      *
      */

    public TermsAndConditions createTermsAndConditions(TermsAndConditions input)
;
     /**
      * 
      *
      */

    public Optional<TermsAndConditions> updateTermsAndConditions(Long id, TermsAndConditions input)
;
     /**
      * 
      *
      */

    public Optional<TermsAndConditions> getCurrentTermsAndConditions(String lang)
;
     /**
      * 
      *
      */

    public void acceptTermsAndConditions(AcceptedTermsAndConditionsInput input)
;


}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/termsandconditions/TermsAndConditionsService.kt
package io.zenwave360.example.clinicaltool.modules.termsandconditions

import io.zenwave360.example.clinicaltool.modules.termsandconditions.domain.*
import io.zenwave360.example.clinicaltool.modules.termsandconditions.dtos.*
import java.math.*
import java.time.*

/** Inbound Service Port for managing [AcceptedTermsAndConditions, TermsAndConditions]. */
@org.springframework.modulith.NamedInterface("TermsAndConditionsService")
interface TermsAndConditionsService {

    /**  */
    fun listTermsAndConditions(): List<TermsAndConditions>

    /**  */
    fun getTermsAndConditions(id: Long): TermsAndConditions?

    /**  */
    fun createTermsAndConditions(input: TermsAndConditions): TermsAndConditions

    /**  */
    fun updateTermsAndConditions(id: Long, input: TermsAndConditions): TermsAndConditions?

    /**  */
    fun getCurrentTermsAndConditions(lang: String): TermsAndConditions?

    /**  */
    fun acceptTermsAndConditions(input: AcceptedTermsAndConditionsInput): Unit
}
